client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -236,6 +236,9 @@ object Client {
     }
 }
 
-final case class UnexpectedStatus(status: Status) extends RuntimeException with NoStackTrace {
-  override def getMessage: String = s"unexpected HTTP status: $status"
+final case class UnexpectedStatus(status: Status, originalRequest: Uri, method: Method)
```

Comment:
I think `originalRequest` is too broad a name.  And method usually comes before uri in our API (and on the wire).  How about:

```suggestion
final case class UnexpectedStatus(status: Status, requestMethod: Method, requestUri: Uri)
```